src/report/report.controller.ts

```diff
@@ -4,34 +4,43 @@ import { ReportService } from './report.service';
 import { AuthGuard } from '../guards/auth.guard';
 import { SystemReportDTO, UserReportDTO } from './report.dto';
 import { UserId } from '../decorators/user-id.decorator';
+import { RolesGuard } from '../guards/roles.guard';
+import { Roles } from '../decorators/roles.decorator';
+import { UserRole } from '../enum/user.enum';
+import { ReportType } from '../enum/report.enum';
 
 @ApiBearerAuth()
 @ApiTags('Report')
-@UseGuards(AuthGuard)
+@UseGuards(AuthGuard, RolesGuard)
 @Controller('report')
 export class ReportController {
     constructor(private readonly service: ReportService) {}
 
+    @Roles(UserRole.Admin)
     @Get()
-    getAllReports() {
-        return this.service.getAllReports();
+    find() {
+        return this.service.find();
     }
 
+    @Roles(UserRole.Admin)
     @Get('user')
-    getUserReports() {
-        return this.service.getUserReports();
+    findUserReports() {
+        return this.service.find({ type: ReportType.User });
     }
 
+    @Roles(UserRole.Admin)
     @Get('system')
-    getSystemReports() {
-        return this.service.getSystemReports();
+    findSystemReports() {
+        return this.service.find({ type: ReportType.User });
     }
```

Comment:
is this supposed to be ReportType.System?

Reply:
Yes